activerecord/test/models/cpk/book.rb

```diff
@@ -5,6 +5,7 @@ class Book < ActiveRecord::Base
     self.table_name = :cpk_books
     self.primary_key = [:author_id, :number]
 
+    belongs_to :order, autosave: true, query_constraints: [:shop_id, :order_id]
```

Comment:
```suggestion
    belongs_to :order, query_constraints: [:shop_id, :order_id]
```
Autosave should be enabled by default.